autoPyTorch/pipeline/components/training/trainer/base_trainer.py

```diff
@@ -278,6 +384,12 @@ def train_epoch(self, train_loader: torch.utils.data.DataLoader, epoch: int,
                     epoch * len(train_loader) + step,
                 )
 
+        if self.scheduler:
+            if 'ReduceLROnPlateau' in self.scheduler.__class__.__name__:
+                self.scheduler.step(loss)
+            else:
+                self.scheduler.step()
+
```

Comment:
We need to merge here accordingly (since I made a different PR for it in `development` branch)

Especially, `ReduceLROnPlateau` must be performed after the validation step.